lib/Segments/SegmentSubscribersRepository.php

```diff
@@ -24,12 +25,17 @@ class SegmentSubscribersRepository {
   /** @var FilterHandler */
   private $filterHandler;
 
+  /** @var TransientCache */
+  private $transientCache;
```

Comment:
I am not sure if having the cache here in the repository is a good idea. You can convince me otherwise. I always thought repositories are in the code for SQL queries. I would add another layer on top of the repository and deal with the caching in there.